### PR TITLE
Feature/APP-3120_payment_method_selection_second_screen

### PR DESCRIPTION
**Pull request by Gracian0-Caldeir4**
**What does this PR do?**

   This PR adds a dedicated view for browsing and selecting payment methods, enhancing the payment experience by providing users with a focused interface. Upon selecting a payment method, users are seamlessly redirected back to the main payment screen with their choice pre-selected, aligning with best practices for clarity and ease of use in the payment process. 

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] BottomSheetView.swift
- [ ] PurchaseBottomSheet.swift
- [ ] SelectedPaymentMethodView.swift
- [ ] PaymentMethodChoiceView.swift
- [ ] BottomSheetViewModel.swift

**How should this be manually tested?**

  Run the Trivial Drive feature and tap on the “Select Payment Method” button. Choose a payment method and verify that the flow correctly returns to the main bottom sheet with the selected payment method pre-selected.

**What are the relevant tickets?**

  Tickets related to this pull-request: APP-3120

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

**Review by tiaguu — CHANGES_REQUESTED** [review]
Overall, it is quite good. Just some matters regarding variable naming that need to be adjusted.

File: Sources/AppCoinsSDK/Localization/Localizable.xcstrings [hunk 1]
```diff
@@ -1539,6 +1539,9 @@
           }
         }
       }
+    },
+    "chevron.right" : {
```
[the remark this answers is not in the document]
Comment (Gracian0-Caldeir4) [reply Sources/AppCoinsSDK/Localization/Localizable.xcstrings]: Done

File: Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift [hunk 2]
```diff
@@ -54,6 +54,10 @@ internal class BottomSheetViewModel: ObservableObject {
     
     @Published var isCreditCardView: Bool = false
     
+    @Published var canChooseMethod: Bool = false
```
[the remark this answers is not in the document]
Comment (Gracian0-Caldeir4) [reply Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift]: Done

File: Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift [hunk 3]
```diff
@@ -91,6 +95,10 @@ internal class BottomSheetViewModel: ObservableObject {
         }
     }
     
+    internal func setCanChooseMethod(canChooseMethod: Bool) {
```
[the remark this answers is not in the document]
Comment (Gracian0-Caldeir4) [reply Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift]: Done

File: Sources/AppCoinsSDK/Utils/Constants.swift [hunk 4]
```diff
@@ -22,6 +22,9 @@ internal struct Constants {
     static internal let chooseCard = NSLocalizedString("choose_card_title", bundle: .APPCModule, comment: "Choose a Card")
     static internal let addCard = NSLocalizedString("add_card_button", bundle: .APPCModule, comment: "Add new card")
     static internal let yourCard = NSLocalizedString("your_card_title", bundle: .APPCModule, comment: "Your card")
+    static internal let selectPaymentMethodText = NSLocalizedString("select_payment_method_button", bundle: .APPCModule, comment: "Select payment method")
+    static internal let signToGetBonusText = NSLocalizedString("sign_to_get_bonus_button", bundle: .APPCModule, comment: "Sign in to get your bonus")
```
[the remark this answers is not in the document]
Comment (Gracian0-Caldeir4) [reply Sources/AppCoinsSDK/Utils/Constants.swift]: Done

File: Sources/AppCoinsSDK/UI/Purchase/Cells/PaymentMethodList.swift [hunk 5]
```diff
@@ -21,6 +21,9 @@ internal struct RadioButtonGroupView: View {
                         if (!option.disabled) {
                             transactionViewModel.selectPaymentMethod(paymentMethod: option)
                         }
+                        if viewModel.canChooseMethod == true {
```
[the remark this answers is not in the document]
Comment (Gracian0-Caldeir4) [reply Sources/AppCoinsSDK/UI/Purchase/Cells/PaymentMethodList.swift]: Done

File: Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift [hunk 6]
```diff
@@ -54,6 +54,10 @@ internal class BottomSheetViewModel: ObservableObject {
     
     @Published var isCreditCardView: Bool = false
     
+    @Published var canChooseMethod: Bool = false
+    
+    @Published var didLogin: Bool = false
```
[the remark this answers is not in the document]
Comment (Gracian0-Caldeir4) [reply Sources/AppCoinsSDK/UI/Purchase/ViewModels/BottomSheetViewModel.swift]: Done

File: Sources/AppCoinsSDK/UI/Purchase/PurchaseStateViews/PaymentMethodChoiceView.swift [hunk 7]
```diff
@@ -0,0 +1,122 @@
+//
+//  PaymentMethodChoiceView.swift
+//
+//
+//  Created by Graciano Caldeira on 04/10/2024.
+//
+
+import SwiftUI
+
+struct PaymentMethodChoiceView: View {
```
[the remark this answers is not in the document]
Comment (Gracian0-Caldeir4) [reply Sources/AppCoinsSDK/UI/Purchase/PurchaseStateViews/PaymentMethodChoiceView.swift]: Done

**Pull request merged**